cli/command/registry/logout.go

```diff
@@ -53,6 +54,13 @@ func runLogout(_ context.Context, dockerCli command.Cli, serverAddress string) e
 		regsToLogout = append(regsToLogout, hostnameAddress, "http://"+hostnameAddress, "https://"+hostnameAddress)
 	}
 
+	if isDefaultRegistry {
```

Comment:
Don't think this is new, but we're now using it twice; wondering if we're missing a new check for `isDefaultRegistry` after the above branch does a `credentials.ConvertToHostname(serverAddress)`


(unrelated: I think the intent of `ConvertToHostname` is effectively "normalise storage key" - we need to look at some of that at some point - naming is confusing, and not always properly describes intent)

Reply:
> On another look, I guess a user could try to logout by doing docker logout index.docker.io and we should make that work.

Yes,  that's roughly the scenario I had in mind. Technically it could go even further (log out for `[http(s)://]registry-1.docker.io` etc), but it looks like the current code only assumes "no registry" passed to mean docker hub. For the old code that made no real difference, as in that case it would at most do too many logouts (`hostnameAddress, "http://"+hostnameAddress, "https://"+hostnameAddress`), but for the new flow, it may be relevant that we would miss the `manager.NewManager(store).Logout()` call.

Reply:
So I started looking into this, and the current situation is – the web-based flow only triggers when the user runs either:
- `docker login`, or
- `docker login https://index.docker.io/v1/`
the following do not work:
- `docker login index.docker.io`
- `docker login index.docker.io/v1/
- `docker login https://index.docker.io`

For logout, the following trigger an oauth logout:
- `docker logout`, or
- `docker logout https://index.docker.io/v1/`

This seems consistent to me, but WDYT @thaJeztah?